packages/api-reference/src/features/DocumentSource/hooks/useDocumentSource.ts

```diff
@@ -0,0 +1,183 @@
+import { createActiveEntitiesStore, createWorkspaceStore } from '@scalar/api-client/store'
+import { dereference, normalize, upgrade } from '@scalar/openapi-parser'
+import type { OpenAPI, OpenAPIV3_1 } from '@scalar/openapi-types'
+import { type ApiReferenceConfiguration, apiReferenceConfigurationSchema } from '@scalar/types/api-reference'
+import { type MaybeRefOrGetter, type Ref, computed, ref, toValue, watch } from 'vue'
+
+import { parse } from '@/helpers/parse'
+import { useSidebar } from '@/hooks/useSidebar'
+import { createEmptySpecification } from '@/libs/openapi'
+
+import { measure } from '@/helpers/measure'
+import type { Spec } from '@scalar/types/legacy'
+import { useDocumentFetcher } from './useDocumentFetcher'
+
+/**
+ * Pass any data source, retrieve all the data stores and whatever we need to render the API reference.
+ */
+export function useDocumentSource({
```

Comment:
The issue with putting these things in a hook is if you call the hook again it will duplicate everything? Thats why its nice to have these things as high up as possible in a component, you don't need to worry about someone calling it again